tests/gcode/test_G20_G21.py

```diff
@@ -11,14 +11,12 @@ def test_gcodes_G20(self):
     self.assertEqual(self.printer.unit_factor, 25.4)
 
   def test_G20_is_not_buffered(self):
-    g = Gcode({"message": "G20"})
-    self.assertFalse(self.printer.processor.is_buffered(g))
```

Comment:
That sounds like an improvement, not a problem with this PR?

Reply:
My form of addressing the concern can reasonably be considered an improvement. to the existing code.  However, my objection to your proposed modification to the tests is that you are changing the wrong thing in the first place. It is not the test that is at fault, but rather that the code being tested fails. Your proposed change is, in effect, saying "Well, we just won't test that", but hide that decision by changing the test to appear on the surface as if it is doing something useful.

Reply:
Yes, we won't test that because it doesn't exist and hasn't for some time.
GCodeProcessor.is_buffered() was removed in https://github.com/intelligent-agent/redeem/commit/f2e58ed60a01de7ef6bf0664acd47f15175416ae

And even then, it did this:
```
    def is_buffered(self, gcode):
        val = gcode.code()
        if not val in self.gcodes:
            return False
        return self.gcodes[val].is_buffered()
```
So yes, I believe changing `self.printer.processor.is_buffered(g)` to `self.printer.processor.gcodes["G20"].is_buffered()` exactly preserves the intent of the test because it exactly matches the changes made to GCodeProcessor.

Reply:
You seem to be missing my point. ```g = Gcode({"message": "G20"})``` is the way we access a code to perform the G20 action. So, for the test to be valid, you must use that mechanism to generate your test instance. It is quite possible that ` ...processor.is_buffered(g)` is not the appropriate check. I suspect that `g.is_buffered()` has replaced it. But, the point is that you are proposing to test a g-like object from a different source. (They may actually be the same, but for the sake of meaningful testing, you cannot assume that is true.)

Reply:
You seem to have a very different testing philosophy than the term "unit test" actually describes. In the test I've written, the G20 GCode processor tests test the G20 GCode processor. In the code you're describing, the G20 GCode processor tests need to test the entire GCode parsing code path or they're somehow invalid.

I want to get this checked in so I can continue some work to simplify/expand the NativePathPlanner APIs. Do you believe this issue should block that, or are you alright with revising this as you see fit in a followup commit?

Reply:
I believe I have addressed this issue with the last change. Do you agree?